pkg/frontend/querymiddleware/codec.go

```diff
@@ -369,14 +398,17 @@ func (c prometheusCodec) EncodeResponse(ctx context.Context, res Response) (*htt
 		sp.LogFields(otlog.Int("series", len(a.Data.Result)))
 	}
 
+	// TODO: select format based on Accept header
```

Comment:
Note to reviewers: this is where we'd implement content negotiation to send the protobuf format to rulers.